### ISSUE TITLE
`StructureOverlay` and dyed wool colors

### ISSUE THREAD

**Issue by Ps0ke** [overviewer/Minecraft-Overviewer#1767]
I am building a subway system on my java edition server and would like to show different lines colored differently automatically on the overviewer. This shall be done by placing the rails on top of dyed wool in the color of the respective line.

Right now I am using the following overlay for the blue line (the only one so far)
```python
def subwayLine(color):
    block = {
        "rail": 66,
        "wool_blue": 35, # :11
        }
    rgb = {
        "blue": (0, 107, 254, 255),
        }
    return (((0,0,0, block['rail']), (0,-1,0, block['wool_'+color])), rgb[color])

SubwayOverlay = StructureOverlay(structures=[subwayLine('blue'),])
```

This would allow for easy addition of new lines. Unfortunately, looking up [`mc_id.h`](https://github.com/overviewer/Minecraft-Overviewer/blob/master/overviewer_core/src/mc_id.h) there is only one block ID for wool and the color is apparently added as some sort of metadata. Unfortnuately I wasn't able to find numerical IDs, just [namespaced IDs](https://minecraft.gamepedia.com/Wool#ID) on the official wiki. According to [this site](https://minecraftitemids.com/item/blue-wool) there are data values ranging 0–13 for the colors, but this might refer to the bedrock edition. Either way, there doesn't seem a way to pass this to the `StructureOverlay`. 

Looking at [`primitives/overlay-structure.c:182`](https://github.com/overviewer/Minecraft-Overviewer/blob/master/overviewer_core/src/primitives/overlay-structure.c#L182), the block is only identified by a block ID, which is more or less just an `int`. I guess adding metadata support for wool colors might be a bigger change. [`world.py:373`](https://github.com/overviewer/Minecraft-Overviewer/blob/master/overviewer_core/world.py#L373) lists the IDs and data values as tuples, maybe this is a way of supporting it?

Am I missing something here? How can I identify wool of different dye colors within a Structure overlay.

In case it's not, I might be interesting helping to implement it. I am pretty good with Python and know the basics of C, so I would need some guidance, especially since I've never contributed to the overviewer before.

**Comment by DragonDev1906** [comment]
@Ps0ke 
Minecraft previously used these data values from 0-15 ([Wiki](https://minecraft.gamepedia.com/Wool#From_block_loot)) but changed to using Strings ("minecraft:white_wool"). Overviewer still uses blockids+datavalues and maps these using  [world.py:373](https://github.com/overviewer/Minecraft-Overviewer/blob/master/overviewer_core/world.py#L373).

With the addition of data-values to StructureOverlay you will be able to use these as shown in my example.

**Comment by enaut** [comment]
Yes when I implemented this I excluded the data values since that was my first Python←→ C interoperability and the last since... furthermore I'm not a C developer... So I was happy that I achieved my goal. Of course this can be the foundation to add wool color wood color etc. so feel free to add the differentiation - I won't do so since I hardly play minecraft anymore.
